sssom/util.py

```diff
@@ -657,9 +658,14 @@ def deal_with_negation(df: pd.DataFrame) -> pd.DataFrame:
             PREDICATE_MODIFIER
         ].fillna("")
 
+    # .fillna(df) towards the end fills an empty value
+    # with a corresponding value from df.
+    # This needs to happen because the columns in df
+    # not in reconciled_df_subset will be NaN otherwise
+    # which is incorrect.
     reconciled_df = df.merge(
         reconciled_df_subset, how="right", on=list(reconciled_df_subset.columns)
-    ).fillna("")
+    ).fillna(df)
```

Comment:
What does thta syntax mean (asking out of curiosity)

Reply:
when we do a `right` join where the columns in the left DataFrame do not exist in the right DataFrame, the corresponding columns of the resultant DataFrame are `NaN` in spite of the left DataFrame for that row having data. The `df` basically fills the resultant DataFrame with appropriate data from the left one.